src/NServiceBus.Core/Transports/Msmq/MsmqConfigurationExtensions.cs

```diff
@@ -18,17 +19,23 @@ public static class MsmqConfigurationExtensions
         /// In some cases it may be useful to use the <see cref="Headers.ControlMessageHeader"/> key to determine if a message is a control message.
         /// The only exception to this rule is received messages with corrupted headers. These messages will be forwarded to the error queue with no label applied.
         /// </remarks>
-        public static void ApplyLabelToMessages(this TransportExtensions<MsmqTransport> transportExtensions, MsmqLabelGenerator labelGenerator)
+        public static TransportExtensions<MsmqTransport> ApplyLabelToMessages(this TransportExtensions<MsmqTransport> transportExtensions, MsmqLabelGenerator labelGenerator)
         {
             Guard.AgainstNull(nameof(labelGenerator), labelGenerator);
             transportExtensions.Settings.Set<MsmqLabelGenerator>(labelGenerator);
+            return transportExtensions;
         }
 
-        internal static MsmqLabelGenerator GetMessageLabelGenerator(this ReadOnlySettings settings)
+        /// <summary>
+        /// Allows the IsolationLevel and transaction timeout to be changed for the TransactionScope used to receive messages. 
```

Comment:
Did you mean:

"Allows to change the transaction isolation level and timeout for the `TransactionScope` used to receive messages."

?

This indeed sounds a little bit better.